### PR TITLE
fix: band mask generation with bands outside the HSI wavelength range

### PR DESCRIPTION
**Pull request by Fersoil**
# Contributing to Meteors

Thank you for contributing to meteors! Please follow the guidelines below to ensure a smooth and efficient contribution process.

## PR Checklist

- [x] **PR Title**: `"semantic tag: description"`

  - **Package**: Use the appropriate semantic tag for your PR:
    - `feat`: A new feature.
    - `fix`: A bug fix.
    - `docs`: Documentation only changes.
    - `style`: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc).
    - `refactor`: A code change that neither fixes a bug nor adds a feature.
    - `perf`: A code change that improves performance.
    - `test`: Adding missing tests or correcting existing tests.
    - `chore`: Changes to the build process or auxiliary tools and libraries such as documentation generation.
    - `build`: Changes that affect the build system or external dependencies.
  - **Example**: `"fix: typo in README.md"`

- [x] **Related Issue(s)**: If your PR fixes an issue, please link it in the PR description.

- [x] **PR Message**: **_Delete this entire checklist_** and replace it with:

  - **Description**: A detailed description of the change.
  - **Dependencies**: Any dependencies required for this change.

- [x] **Add Tests and Docs**: If adding a new features, please include:

  1. A test for the feature, preferably unit tests that do not rely on network access.
  2. An example notebook showing its use, located in the `examples` directory.

- [x] **Lint**: Ensure your code passes linting and all tests:

  - Code is formatted with `pre-commit` hooks. Run `pre-commit install` and `pre-commit run --all-files` to format your code.

- [x] **Tests**: Ensure all tests pass:
  - Run `pytest` in the root directory to run all tests.

## Additional Guidelines

- Import optional dependencies within functions.
- Avoid adding dependencies to `pyproject.toml` files unless required for unit tests.
- Most PRs should not modify more than one package.
- Ensure changes are backwards compatible.

## Review Process

If no one reviews your PR within a few days, please `@-mention` one of the maintainers.

### CODE REVIEW

File: src/meteors/visualize/lime_visualize.py [hunk 1]
```diff
@@ -22,6 +22,31 @@
...
+    Returns:
+        dict[str, int]: A dictionary mapping band names to segment IDs.
+    """
+
+    # simplify bands with multiple segments
+    segments_to_be_updated = []
+
+    for label in band_names.keys():
+        if isinstance(label, tuple) or isinstance(label, list):
+            new_key = ", ".join(label)
```
Comment (WolodjaZ) [inline src/meteors/visualize/lime_visualize.py:40]: Not needed the `, ` space after the coma

**Review by WolodjaZ — APPROVED** [review]
I added one more test for `_merge_band_names_segments`

File: src/meteors/lime.py [hunk 2]
```diff
@@ -138,25 +138,27 @@ def validate_shapes(attributes: torch.Tensor, hsi: HSI) -> None:
     Raises:
         ValueError: If the shape of the attributes tensor does not match the shape of the hsi.
     """
-    if attributes.shape != hsi.image.shape:
+    if list(attributes.shape) != list(hsi.image.shape):
```
[the remark this answers is not in the document]
Comment (Fersoil) [reply src/meteors/lime.py:141]: I noticed that in some cases, yes. The same torch tensor sizes were not correctly compared

File: src/meteors/visualize/lime_visualize.py (hunk 1 above)
Comment (Fersoil) [reply src/meteors/visualize/lime_visualize.py]: sure, I added it for clarity

Comment (Fersoil) [inline src/meteors/visualize/lime_visualize.py]:
```suggestion
            new_key = ",".join(label)
```

File: src/meteors/visualize/lime_visualize.py [hunk 3]
```diff
@@ -22,6 +22,31 @@
...
+            segments_to_be_updated.append((label, new_key))
+
+    # update the dict
+    for old_key, new_key in segments_to_be_updated:
+        band_names[new_key] = band_names.pop(old_key)
```
[the remark this answers is not in the document]
Comment (Fersoil) [reply src/meteors/visualize/lime_visualize.py]: this part of code causes much more updates, there should be only list values filtered, `segments_to_be_updated` does not need to contain the string keys

**Pull request merged**